### ISSUE TITLE
.netrc authentication?

### ISSUE THREAD

**Issue by dnwe** [Michael-F-Bryan/linkcheck#9]
👋 thanks for the great library (consuming via mdbook)

Might it be possible to read credentials for external sites via a local `.netrc` file (e.g.,) and then pass those to reqwest to use as basic auth? This is for testing links to external sites that require authentication

**Comment by Michael-F-Bryan** [comment]
I've never heard of `.netrc` before now. Do you know how it works or if there are existing crates for using them?

**Comment by dnwe** [comment]
@Michael-F-Bryan yes it's fairly trivial to parse as it has a well-defined format https://everything.curl.dev/usingcurl/netrc

There are crates available, e.g., https://docs.rs/netrc/0.4.1/netrc/struct.Netrc.html
